spec.bs

```diff
@@ -1509,18 +1521,46 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
     to the result of running [=is debugging only in cooldown or lockout=] with |buyer|.
   1. [=map/For each=] |signalsUrl| → |perSignalsUrlGenerator| of |perSlotSizeQueryParam|:
     1. [=map/For each=] |slotSizeQueryParam| → |perSlotSizeQueryParam| of |perBuyerGenerator|:
+      1. Let |allTrustedBiddingSignals| be an [=ordered map=] whose [=map/keys=] are [=strings=],
+        and [=map/values=] are [=strings=].
       1. Let |keys| be a new [=ordered set=].
       1. Let |igNames| be a new [=ordered set=].
+      1. Let |lengthLimit| be the maximum value that a {{long}} integer can hold.
       1. Let |fetchSignalStartTime| be |settings|'s [=environment settings object/current monotonic time=].
       1. [=map/For each=] joiningOrigin → |groups| of |perSignalsUrlGenerator|:
         1. [=list/For each=] |ig| of |groups|:
-          1. [=set/Append=] |ig|'s [=interest group/trusted bidding signals keys=] to |keys|.
-          1. [=set/Append=] |ig|'s [=interest group/name=] to |igNames|.
-      1. Let |biddingSignalsUrl| be the result of [=building trusted bidding signals url=] with
-        |signalsUrl|, |keys|, |igNames|, |buyerExperimentGroupId|, |topLevelOrigin|, and
-        |slotSizeQueryParam|.
-      1. Let « |allTrustedBiddingSignals|, |dataVersion| » be the result of [=fetching trusted signals=]
-        with |biddingSignalsUrl| and true.
+          1. Let |putativeKeys| be a [=set/clone=] of |keys|.
+          1. Let |putativeIgNames| be a [=set/clone=] of |igNames|.
+          1. Let |putativeLengthLimit| be  |ig|'s [=interest group/max trusted bidding signals url length=].
+          1. If |ig|'s [=interest group/max trusted bidding signals url length=] is 0 or `&gt;`
+            |lengthLimit|, then set |putativeLengthLimit| to |lengthLimit|.
+          1. [=list/Extend=] |putativeKeys| with |ig|'s [=interest group/trusted bidding signals keys=].

```

Comment:
@qingxinwu, should we also change this Extend -> Append?

Reply:
I think no, since [=interest group/trusted bidding signals keys=] is a list, and we're appending its elements to |putativeKeys|